src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1724,104 +1726,294 @@ private BoundExpression BindAssignment(AssignmentExpressionSyntax node, Diagnost
         /// </summary>
         private BoundExpression BindDeconstructionAssignment(AssignmentExpressionSyntax node, DiagnosticBag diagnostics)
         {
-            SeparatedSyntaxList<ArgumentSyntax> arguments = ((TupleExpressionSyntax)node.Left).Arguments;
-
-            // receiver for Deconstruct
+            // receiver for first Deconstruct step
             var boundRHS = BindValue(node.Right, diagnostics, BindValueKind.RValue);
 
-            int numElements = arguments.Count;
-            Debug.Assert(numElements >= 2); // this should not have parsed as a tuple.
+            SeparatedSyntaxList<ArgumentSyntax> arguments = ((TupleExpressionSyntax)node.Left).Arguments;
+            ImmutableArray<BoundExpression> checkedVariables = BindDeconstructionVariables(arguments, diagnostics);
 
-            // bind the variables and check they can be assigned to
-            var checkedVariablesBuilder = ArrayBuilder<BoundExpression>.GetInstance(numElements);
-            foreach (var argument in arguments)
+            if ((object)boundRHS.Type == null)
             {
-                var boundVariable = BindExpression(argument.Expression, diagnostics, invoked: false, indexed: false);
-                var checkedVariable = CheckValue(boundVariable, BindValueKind.Assignment, diagnostics);
+                if (boundRHS.Kind == BoundKind.TupleLiteral)
+                {
+                    // tuple literal without type such as `(null, null)`, let's fix it up by peeking at the LHS
+                    TypeSymbol lhsAsTuple = MakeTupleTypeFromDeconstructionLHS(checkedVariables, diagnostics, Compilation);
+                    boundRHS = GenerateConversionForAssignment(lhsAsTuple, boundRHS, diagnostics);
+                }
+                else
+                {
+                    // expression without type such as `null`
+                    Error(diagnostics, ErrorCode.ERR_DeconstructRequiresExpression, node);
```

Comment:
Got that now. It is used for location and not as an argument of the message.

---
In reply to: [65793066](https://github.com/dotnet/roslyn/pull/11715#discussion_r65793066) [](ancestors = 65793066)